src/java/org/apache/cassandra/db/virtual/QueriesTable.java

```diff
@@ -0,0 +1,92 @@
+import org.apache.cassandra.schema.TableMetadata;
+
+import static java.lang.Long.max;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static org.apache.cassandra.utils.MonotonicClock.Global.approxTime;
+
+/**
+ * Virtual table that lists currently running queries on the NTR (coordinator) and Read/Mutation (local) stages
+ *
+ * Example:
+ * <pre>
+ * cqlsh> SELECT * FROM system_views.queries;
+ *
+ *  thread_id                   | queued_micros |  running_micros | task
+ * ------------------------------+---------------+-----------------+--------------------------------------------------------------------------------
+ *  Native-Transport-Requests-7 |         72923 |            7611 |                      QUERY select * from system_views.queries; [pageSize = 100]
+ *              MutationStage-2 |         18249 |            2084 | Mutation(keyspace='distributed_test_keyspace', key='000000f8', modifications...
+ *                  ReadStage-2 |         72447 |           10121 |                                         SELECT * FROM keyspace.table LIMIT 5000
+ * </pre>
+ */    
+public class QueriesTable extends AbstractVirtualTable
+{
+    private static final String TABLE_NAME = "queries";
+    private static final String ID = "thread_id";
+    private static final String QUEUED = "queued_micros";
+    private static final String RUNNING = "running_micros";
+    private static final String DESC = "task";
+
+    QueriesTable(String keyspace)
+    {
+        super(TableMetadata.builder(keyspace, TABLE_NAME)
+                           .kind(TableMetadata.Kind.VIRTUAL)
+                           .partitioner(new LocalPartitioner(UTF8Type.instance))
+                           // The thread name is unique since the id given to each SEPWorker is unique
+                           .addPartitionKeyColumn(ID, UTF8Type.instance)
+                           .addRegularColumn(QUEUED, LongType.instance)
+                           .addRegularColumn(RUNNING, LongType.instance)
+                           .addRegularColumn(DESC, UTF8Type.instance)
+                           .build());
+    }
+
+    /**
+     * Walks the {@link SharedExecutorPool} workers for any {@link DebuggableTask} instances and populates the table.
+     */
+    @Override
+    public DataSet data()
+    {
+        SimpleDataSet result = new SimpleDataSet(metadata());
+        long now = approxTime.now();
+        
+        for (DebuggableTask.RunningDebuggableTask task : SharedExecutorPool.SHARED.runningTasks())
+        {
+            if (!task.hasTask()) continue;
+            
+            long creationTimeNanos = task.creationTimeNanos();
+            long startTimeNanos = task.startTimeNanos();
+            long queuedMicros = NANOSECONDS.toMicros(max((startTimeNanos > 0 ? startTimeNanos : now) - creationTimeNanos, 0));
+            long runningMicros = startTimeNanos > 0 ? NANOSECONDS.toMicros(now - startTimeNanos) : 0;
```

Comment:
In the original patch, this looked like..

```
long runningMicros = approxTimeOfStart > 0 ? NANOSECONDS.toMicros(max(approxTimeOfStart - now, 0)) : 0;
```

It seemed like the `approxTimeOfStart - now` was reversed, and the max wasn't necessary if `now` should never be less than the start time. Although thinking more about it, we should probably just re-acquire `now` for every task...

CC @clohfink